src/sentry_core.c

```diff
@@ -364,6 +374,18 @@ sentry_options_get_dsn(const sentry_options_t *opts)
     return opts->raw_dsn;
 }
 
+void
+sentry_options_set_sample_rate(sentry_options_t *opts, double sample_rate)
+{
+    opts->sample_rate = sample_rate;
```

Comment:
_nit_: Maybe normalize it to `(0.0, 1.0)` so that `get_sample_rate` always returns a valid rate?